Common/Tools/TrackTuner.h

```diff
@@ -295,330 +349,378 @@ struct TrackTuner {
+    std::string grOneOverPtPionNameData = "sigmaVsPtData";
+    grOneOverPtPionMC.reset(dynamic_cast<TGraphErrors*>(inputFileQoverPt->Get(grOneOverPtPionNameMC.c_str())));
+    grOneOverPtPionData.reset(dynamic_cast<TGraphErrors*>(inputFileQoverPt->Get(grOneOverPtPionNameData.c_str())));
   }
 
   template <typename T1, typename T2, typename T3, typename T4, typename H>
   void tuneTrackParams(T1 const& mcparticle, T2& trackParCov, T3 const& matCorr, T4 dcaInfoCov, H hQA)
   {
-
     double ptMC = mcparticle.pt();
+    double dcaXYResMC = 0.0; // sd0rpo=0.;
+    double dcaZResMC = 0.0;  // sd0zo =0.;
 
-    double dcaXYResCurrent = 0.0; // sd0rpo=0.;
-    double dcaZResCurrent = 0.0;  // sd0zo =0.;
-
-    double dcaXYResUpgr = 0.0; // sd0rpn=0.;
-    double dcaZResUpgr = 0.0;  // sd0zn =0.;
-
-    // double OneOverPtCurrent = 0.0; // spt1o =0.;
-    // double OneOverPtUpgr = 0.0; // spt1n =0.;
-
-    double dcaXYMeanCurrent = 0.0; // sd0mrpo=0.;
-    double dcaXYMeanUpgr = 0.0;    // sd0mrpn=0.;
+    double dcaXYResData = 0.0; // sd0rpn=0.;
+    double dcaZResData = 0.0;  // sd0zn =0.;
 
-    double dcaXYPullCurrent = 1.0;
-    double dcaXYPullUpgr = 1.0;
+    double dcaXYMeanMC = 0.0;   // sd0mrpo=0.;
+    double dcaXYMeanData = 0.0; // sd0mrpn=0.;
 
-    double dcaZPullCurrent = 1.0;
-    double dcaZPullUpgr = 1.0;
+    double dcaXYPullMC = 1.0;
+    double dcaXYPullData = 1.0;
 
-    dcaXYResCurrent = evalGraph(ptMC, grDcaXYResVsPtPionCurrent.get());
-    dcaXYResUpgr = evalGraph(ptMC, grDcaXYResVsPtPionUpgr.get());
+    double dcaZPullMC = 1.0;
+    double dcaZPullData = 1.0;
 
-    // dcaXYResCurrent = 1.0;
-    // dcaXYResUpgr = 1.5;
+    dcaXYResMC = evalGraph(ptMC, grDcaXYResVsPtPionMC.get());
+    dcaXYResData = evalGraph(ptMC, grDcaXYResVsPtPionData.get());
 
-    dcaZResCurrent = evalGraph(ptMC, grDcaZResVsPtPionCurrent.get());
-    dcaZResUpgr = evalGraph(ptMC, grDcaZResVsPtPionUpgr.get());
+    dcaZResMC = evalGraph(ptMC, grDcaZResVsPtPionMC.get());
+    dcaZResData = evalGraph(ptMC, grDcaZResVsPtPionData.get());
 
-    // dcaZResCurrent = 1.0;
-    // dcaZResUpgr = 1.0;
-
-    // OneOverPtCurrent = evalGraph(ptMC, grOneOverPtPionCurrent.get() );
-    // OneOverPtUpgr = evalGraph(ptMC, grOneOverPtPionUpgr.get() );
-
-    // OneOverPtCurrent = 1.0;
-    // OneOverPtUpgr = 2.0;
+    // For Q/Pt corrections, files on CCDB will be used if both qOverPtMC and qOverPtData are null
+    if ((qOverPtMC == 0) && (qOverPtData == 0)) {
+      qOverPtMC = evalGraph(ptMC, grOneOverPtPionMC.get());
+      qOverPtData = evalGraph(ptMC, grOneOverPtPionData.get());
+    }
 
-    dcaXYMeanCurrent = evalGraph(ptMC, grDcaXYMeanVsPtPionCurrent.get());
-    dcaXYMeanUpgr = evalGraph(ptMC, grDcaXYMeanVsPtPionUpgr.get());
+    if ((qOverPtMC < 0) || (qOverPtData < 0)) {
+      LOG(fatal) << " Please assign a positive integer to qOverPtMC and qOverPtData in your configuration! Please refer to the trackTuner documentation.";
+    }
 
```

Comment:
```suggestion

    if(updateCurvature && updateCurvatureIU) {
        if ((qOverPtMC < 0) || (qOverPtData < 0)) {
          LOG(info) << "### q/pt smearing: qOverPtMC=" << qOverPtMC << ", qOverPtData=" << qOverPtData << ". One of them is negative. Retrieving then values from graphs from input .root file";
          qOverPtMC = evalGraph(ptMC, grOneOverPtPionMC.get());
          qOverPtData = evalGraph(ptMC, grOneOverPtPionData.get());
        }
    }
```

Reply:
This suggestion from my side should solve the issue that I was raising before, i.e. the one related to equalities with `float`s.
I reverted a bit the logic. Here the explanation:

 - first of all, I'm protecting everything with `if(updateCurvature || updateCurvatureIU) {` (if we do not want to smear the pt, then we do not care)
 - if the values are negative, then we query the correction graphs from the .root file, and we throw a fatal in case of issue in retrieving the graphs